src/common/ceph_time.h

```diff
@@ -20,6 +20,19 @@
 
 #include "include/encoding.h"
 
+#if defined(DARWIN)
+#include <sys/_types/_timespec.h>
+#include <mach/mach.h>
+#include <mach/clock.h>
+
+#define CLOCK_REALTIME CALENDAR_CLOCK
```

Comment:
But it would be nice if they all end up at the same place??

This the discussion with arrisen quite a feew times now.
Most programmes still asume that they are programming against Linux.
And I think it is less error prone if we stick it in compat.h so it is fixed for all architectures/OSes at once.
The advantage being that all this CLOCK* defines are in one place.